source/fundamentals/builders.txt

```diff
@@ -0,0 +1,288 @@
+.. _csharp-builders:
+
+========================
+Operations with Builders
+========================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn about the helper classes, or **builders**, that
+the {+driver-short+} provides to construct various types used in the
+driver. The driver includes types related to the
+specification of filters, updates, projections, sorts, and index keys,
+and you can use builders to aid in the creation of these definitions.
+Using the builder classes, you leverage the power of the
+{+lang-framework+} compiler and your IDE to find errors during
+development and your IDE for discovery and code completion.
```

Comment:
This is a bit confusing, and looking.

I'd like to see a delineated list of the "types" that builders can produce. I also think this paragraph and the following paragraph overlap. Perhaps structure it so the value proposition of builders is clearly stated at the beginning, and then the types the builder produces and how they're used (projection builders build a projection type, but that probably means more to users if stated as they build a projection operation".

Reply:
I broke up the paragraph, deleted some repetitive information, and presented the tasks in a list format for easier comprehension.